subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -498,8 +513,32 @@ static bool broadcast_source_found(struct bt_data *data, void *user_data)
 	char addr_str[BT_ADDR_LE_STR_LEN];
 	struct bt_uuid_16 adv_uuid;
 	uint32_t broadcast_id;
+	char broadcast_name[MAX_BROADCASTNAME_LEN + 1] = { 0 };
+	size_t broadcast_name_length = 0;
+	bool found_matched_broadcast_id = false;
+	bool found_matched_broadcast_name = false;
 	int err;
 
+	bt_addr_le_to_str(info->addr, addr_str, sizeof(addr_str));
+
+	/* Check for matched broadcast name (in case of add_by_broadcast_name) */
+	if (data->type == BT_DATA_BROADCAST_NAME) {
+		if (broadcast_name_scan.broadcast_name[0] != '\0') {
+			broadcast_name_length = (data->data_len < sizeof(broadcast_name)) ?
+				data->data_len : sizeof(broadcast_name) - 1;
+			memcpy(broadcast_name, data->data, broadcast_name_length);
```

Comment:
Broadcast names are UTF-8. Suggest to use `utf8_lcpy` to properly copy a UTF8 string that may be truncated

Reply:
Of course with the fix to the BROADCAST_NAME_LEN macros, we should never need to truncate, and if `data->data_len > sizeof(broadcast_name) - 1` then it's simply invalid and should be ignored

Reply:
OK.